pkg/nodecheckerrunner/runner_test.go

```diff
@@ -0,0 +1,244 @@
+package nodecheckerrunner
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	chmv1alpha1 "github.com/Azure/cluster-health-monitor/apis/chm/v1alpha1"
+	"github.com/Azure/cluster-health-monitor/pkg/checker"
+)
+
+// mockChecker implements the NodeChecker interface for testing
+type mockChecker struct {
+	name   string
+	result *checker.Result
+	err    error
+	calls  int
+}
+
+func (m *mockChecker) Name() string {
+	return m.name
+}
+
+func (m *mockChecker) Run(ctx context.Context) (*checker.Result, error) {
+	m.calls++
+	return m.result, m.err
+}
+
+func TestRunCheckers(t *testing.T) {
+	tests := []struct {
+		name           string
+		checkers       []NodeChecker
+		existingCR     *chmv1alpha1.CheckNodeHealth
+		expectedStatus int // number of results expected
```

Comment:
I don't think expectedStatus is ever checked. This is probably left over from before validateFunc was added because I see that checks the length of the results.

Reply:
you are right. I have removed expectedStatus.